src/config/options.rs

```diff
@@ -468,3 +470,156 @@ pub enum MatchArmLeadingPipe {
     /// Preserve any existing leading pipes
     Preserve,
 }
+
+/// Defines the default values for each config according to [the style guide].
+/// rustfmt output may differ between style editions.
+///
+/// [the style guide]: https://doc.rust-lang.org/nightly/style-guide/
+#[config_type]
+pub enum StyleEdition {
+    #[value = "2015"]
+    #[doc_hint = "2015"]
+    /// [Edition 2015]()
+    Edition2015,
+    #[value = "2018"]
+    #[doc_hint = "2018"]
+    /// [Edition 2018]()
+    Edition2018,
+    #[value = "2021"]
+    #[doc_hint = "2021"]
+    /// [Edition 2021]()
+    Edition2021,
+    #[value = "2024"]
+    #[doc_hint = "2024"]
+    /// [Edition 2024]().
+    Edition2024,
+}
+
+/// Defines unit structs to implement `StyleEditionDefault` for.
+#[macro_export]
+macro_rules! config_option_with_style_edition_default {
+    ($name:ident, $config_ty:ty, $default:expr) => {
+        #[allow(unreachable_pub)]
+        pub struct $name;
+        crate::style_edition_default!($name, $config_ty, $default);
+    };
+    ($name:ident, $config_ty:ty, $default_2015:expr, $default_2024:expr) => {
+        pub struct $name;
+        crate::style_edition_default!($name, $config_ty, $default_2015, $default_2024);
+    };
+    ($($name:ident, $config_ty:ty, $default:expr $(,$default_2024:expr)?);* $(;)*) => {
+        $(
+            config_option_with_style_edition_default!(
+                $name, $config_ty, $default $(,$default_2024)?
+            );
+        )*
+    };
+}
+
+config_option_with_style_edition_default!(
+    // Fundamental stuff
+    MaxWidth, usize, 100;
+    HardTabs, bool, false;
+    TabSpaces, usize, 4;
+    NewlineStyleConfig, NewlineStyle, NewlineStyle::Auto;
+    IndentStyleConfig, IndentStyle, IndentStyle::Block;
```

Comment:
Defines a unit struct for each config option that implements `StyleEditionDefault`